### PR TITLE
[sitecore-jss-rendering-host] Retire devServer

### PR DESCRIPTION
**Pull request by art-alexeyenko**
- [x] This PR follows the [Contribution Guide](https://github.com/Sitecore/jss/blob/dev/CONTRIBUTING.md)
- [x] Changelog updated

## Description / Motivation

devServer is removed from sitecore-jss-rendering-host. renderingHostServer would be the only way to use rendering-host with react apps.
devServer was poorly documented and rarely used which led to the decisions to retire it

## Testing Details
- [ ] Unit Test Added
- [x] Manual Test/Other (Please elaborate)

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -11,6 +11,10 @@ Our versioning strategy is as follows:
 
 ## Unreleased
 
+### 🛠 Breaking Changes
+
+* `[sitecore-jss-rendering-host]` devServer is retired. renderingHostServer is the only way to use rendering-host with react from now on. ([#1426](https://github.com/Sitecore/jss/pull/1426))
```
Comment (ambrauer) [inline CHANGELOG.md:16]: We should use specific function names here to be clear. Also, it's not exclusive to React.
```suggestion
* `[sitecore-jss-rendering-host]` `startDevServer` is retired. `startRenderingHostServer` is the only way to start the rendering host from now on. ([#1426](https://github.com/Sitecore/jss/pull/1426))
```

**Review by ambrauer — APPROVED** [review]
Only a suggestion for the README, otherwise 👍

File: packages/sitecore-jss-rendering-host/package.json [hunk 2]
```diff
@@ -45,6 +46,7 @@
     "@types/webpack-dev-middleware": "2.0.5",
     "@types/webpack-dev-server": "3.1.7",
     "chai": "^4.3.0",
+    "compression": "^1.7.2",
```
Comment (illiakovalenko) [inline packages/sitecore-jss-rendering-host/package.json:49]: I'm curious, but why _compression_ is added to _dependencies_ and _devDependencies_? It should be enough to just include it in _dependencies_

**Review by illiakovalenko — APPROVED** [review]
Looks good to me, see one comment below

File: packages/sitecore-jss-rendering-host/package.json (hunk 2 above)
Comment (art-alexeyenko) [reply packages/sitecore-jss-rendering-host/package.json:49]: Yup, that's an oversight on my part.

**Pull request merged**